core/storage/plugin_unix.go

```diff
@@ -8,20 +8,46 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"path/filepath"
 	"strings"
+	"sync"
 
 	"github.com/docker/go-connections/sockets"
 	"github.com/leodotcloud/log"
 	"github.com/pkg/errors"
 	"github.com/rancher/agent/model"
 )
 
-func CallRancherStorageVolumePlugin(volume model.Volume, action string, payload interface{}) (Response, error) {
+var client = &http.Client{}
+var transportMap = &transportStore{
+	tMap: map[string]*http.Transport{},
+	lock: sync.RWMutex{},
+}
+
+type transportStore struct {
+	tMap map[string]*http.Transport
+	lock sync.RWMutex
+}
+
+func (t *transportStore) add(driver string) {
+	t.lock.Lock()
+	defer t.lock.Unlock()
 	transport := new(http.Transport)
-	sockets.ConfigureTransport(transport, "unix", rancherStorageSockPath(volume))
-	client := &http.Client{
-		Transport: transport,
+	sockets.ConfigureTransport(transport, "unix", filepath.Join(rancherSockDir, driver+".sock"))
+	t.tMap[driver] = transport
+}
+
+func (t *transportStore) get(driver string) *http.Transport {
+	t.lock.RLock()
+	defer t.lock.RUnlock()
+	return t.tMap[driver]
+}
+
+func CallRancherStorageVolumePlugin(volume model.Volume, action string, payload interface{}) (Response, error) {
+	if transportMap.get(volume.Data.Fields.Driver) == nil {
+		transportMap.add(volume.Data.Fields.Driver)
 	}
+	client.Transport = transportMap.get(volume.Data.Fields.Driver)
```

Comment:
i dont think you can have just a single client and swap out its transport and be concurrent safe. What if goroutine A was tring to hit rancher-nfs and set the client's transport to that but then goroutine B rushed in and swapped the transport to point at rancher-ebs before goroutine A could finish its calls?

Maybe the store should store entire clients instead of just transports